make_prg/subcommands/from_msa.py

```diff
@@ -52,95 +75,94 @@ def register_parser(subparsers):
-    )
-    subparser_msa.add_argument("--log", help="Path to write log to. Default is stderr")
+
     subparser_msa.set_defaults(func=run)
 
     return subparser_msa
 
 
-def run(options):
-    MSA_file = Path(options.MSA).resolve()
-    if not MSA_file.exists():
-        raise ValueError(f"File not found: {options.MSA}")
-    output_dir = Path(options.output_dir).resolve()
-    output_dir.mkdir(parents=True, exist_ok=True)
+def get_all_input_files(input_path: str) -> List[Path]:
+    input_path = Path(input_path)
+    if not input_path.exists():
+        raise FileNotFoundError(f"{input_path} does not exist")
 
-    if options.prg_name is None:
-        options.prg_name = MSA_file.stem
-    ofile_prefix = output_dir / options.prg_name
+    if input_path.is_file():
+        all_files = [input_path]
+    else:
+        all_files = [
+            Path(path).absolute() for path in input_path.iterdir() if path.is_file()
+        ]
+    return all_files
 
-    # Set up file logging
-    formatter = logging.Formatter(
-        fmt="%(levelname)s %(asctime)s %(message)s", datefmt="%d/%m/%Y %I:%M:%S"
-    )
-    handler = (
-        logging.FileHandler(options.log) if options.log else logging.StreamHandler()
-    )
-    handler.setFormatter(formatter)
-    logging.getLogger().addHandler(handler)
 
-    logging.info(
-        "Input parameters max_nesting: %d, min_match_length: %d",
-        options.max_nesting,
-        options.min_match_length,
-    )
+def process_MSA(msa_filepath: Path):
+    global options
+    logger.info(f"Generating PRG for {msa_filepath}...")
+    msa_name = msa_filepath.name
+    locus_name = msa_filepath.with_suffix("").name
+
+    temp_dir = io_utils.get_temp_dir_for_multiprocess(options.temp_dir)
+    prefix = str(temp_dir / msa_name)
 
-    prg_fname = ofile_prefix.with_suffix(".prg")
-    if prg_fname.exists() and options.no_overwrite:
-        logging.info(f"Re-using existing prg file {prg_fname}")
-        aseq = prg_builder.PrgBuilder(
-            options.MSA,
+    try:
+        builder = prg_builder.PrgBuilder(
+            locus_name=locus_name,
+            msa_file=msa_filepath,
             alignment_format=options.alignment_format,
             max_nesting=options.max_nesting,
-            min_match_length=options.min_match_length,
-            prg_file=prg_fname,
+            min_match_length=options.min_match_length
         )
+
+        logger.info(f"Writing output files of locus {msa_name}")
+        prg = builder.build_prg()
+        builder.write_prg(prefix, prg)
+        gfa.GFA_Output.write_gfa(prefix, prg)
+        builder.serialize(f"{prefix}.pickle")
+        if options.output_graphs:
+            builder.output_debug_graphs(Path(options.output_prefix + "_debug_graphs"))
+
+    except ValueError as value_error:
+        if "No records found in handle" in value_error.args[0]:
+            logger.warning(f"No records found in MSA {msa_filepath}, skipping...")
+        else:
+            raise value_error
```

Comment:
Do we know what sort of circumstances would cause this to happen? Maybe we could add a logger.error message instead? Or would the python error be informative?

Sorry, I'mm too used to Rust now where Errors are treated as first-class citizens and are very important.

Reply:
Not sure I understand the question... As far as I remember, the only `ValueError` I've got so far is when there is an empty MSA, then `AlignIO.read()` throws a `ValueError` with this specific error message. Just in case we get another type of `ValueError`, we are reraising it to the caller.

I am not sure about changing `logger.warning()` to `logger.error()` as we don't error out on this exception, we simply ignore this MSA and continue execution. The python error/stacktrace is not very informative I think, as we know the circumstances that this exception happens with this error message.

I am not sure I was clear on my answer... Given this, what do you propose us to do here?